src/flint/test/test_all.py

```diff
@@ -2668,19 +2668,46 @@ def setbad(obj, i, val):
 
 def _all_mpolys():
     return [
-        (flint.fmpz_mpoly, flint.fmpz_mpoly_ctx, flint.fmpz, False),
-        (flint.fmpq_mpoly, flint.fmpq_mpoly_ctx, flint.fmpq, True),
+        (flint.fmpz_mpoly, flint.fmpz_mpoly_ctx.get_context, flint.fmpz, False),
+        (flint.fmpq_mpoly, flint.fmpq_mpoly_ctx.get_context, flint.fmpq, True),
+        (
+            flint.fmpz_mod_mpoly,
+            lambda *args, **kwargs: flint.fmpz_mod_mpoly_ctx.get_context(*args, **kwargs, modulus=101),
+            flint.fmpz,
+            True,
+        ),
+        (
+            flint.fmpz_mod_mpoly,
+            lambda *args, **kwargs: flint.fmpz_mod_mpoly_ctx.get_context(*args, **kwargs, modulus=100),
+            flint.fmpz,
+            False,
+        ),
+        (
+            flint.nmod_mpoly,
+            lambda *args, **kwargs: flint.nmod_mpoly_ctx.get_context(*args, **kwargs, modulus=101),
+            int,
+            True,
+        ),
+        (
+            flint.nmod_mpoly,
+            lambda *args, **kwargs: flint.nmod_mpoly_ctx.get_context(*args, **kwargs, modulus=100),
+            int,
```

Comment:
The type here should be `nmod` rather than `int`. The tests should also check for interoperability with `int` as well.

Reply:
I've just pushed some changes that add a lot more interop between `mpoly` types and their base counter part, that should resolve this